src/devices/giolister.cpp

```diff
@@ -94,11 +94,19 @@ void GioLister::Init() {
   g_list_free(mounts);
 
   // Connect signals from the monitor
-  CHECKED_GCONNECT(monitor_, "volume-added", &VolumeAddedCallback, this);
-  CHECKED_GCONNECT(monitor_, "volume-removed", &VolumeRemovedCallback, this);
-  CHECKED_GCONNECT(monitor_, "mount-added", &MountAddedCallback, this);
-  CHECKED_GCONNECT(monitor_, "mount-changed", &MountChangedCallback, this);
-  CHECKED_GCONNECT(monitor_, "mount-removed", &MountRemovedCallback, this);
+  signals_.append(CHECKED_GCONNECT(monitor_, "volume-added", &VolumeAddedCallback, this));
+  signals_.append(CHECKED_GCONNECT(monitor_, "volume-removed", &VolumeRemovedCallback, this));
+  signals_.append(CHECKED_GCONNECT(monitor_, "mount-added", &MountAddedCallback, this));
+  signals_.append(CHECKED_GCONNECT(monitor_, "mount-changed", &MountChangedCallback, this));
+  signals_.append(CHECKED_GCONNECT(monitor_, "mount-removed", &MountRemovedCallback, this));
+}
+
+GioLister::~GioLister()
```

Comment:
Please be consistent with the other part of the file:
GioLister::~GioLister() {
instead of
GioLister::~GioLister()
{

Same for the loop